roles/rsyslog/tasks/main.yaml

```diff
@@ -1,41 +1,47 @@
 ---
-- name: Install/Update required packages
+- name: Install/Update required basic packages
   yum:
-    name: '{{ item }}'
+    name: '{{rsyslog_base_packages|flatten + rsyslog_packages|flatten}}'
     state: 'latest'
-  with_flattened:
-    - '{{ rsyslog__base_packages }}'
-    - '{{ rsyslog__viaq_prereq_packages if rsyslog__viaq|bool else [] }}'
-    - '{{ rsyslog__viaq_packages if rsyslog__viaq|bool else [] }}'
-    - '{{ rsyslog__tls_packages if rsyslog__pki|bool else [] }}'
-    - '{{ rsyslog__packages }}'
-  when: rsyslog__enabled|bool and not use_rsyslog_image|default(False)|bool
+  when: rsyslog_enabled|bool and not use_rsyslog_image|default(False)|bool
+
+- name: Install/Update required pki packages
+  yum:
+    name: '{{rsyslog_tls_packages|flatten}}'
+    state: 'latest'
+  when: rsyslog_enabled|bool and rsyslog_pki|bool and not use_rsyslog_image|default(False)|bool
+
+- name: Install/Update required viaq packages
+  yum:
+    name: '{{rsyslog_viaq_prereq_packages|flatten + rsyslog_viaq_packages|flatten}}'
+    state: 'latest'
+  when: rsyslog_enabled|bool and rsyslog_viaq|bool and not use_rsyslog_image|default(False)|bool
```

Comment:
@sradco , I could not replace it with one task as you suggested (conditions with "rsyslog_viaq|bool" and "rsyslog_pki|bool" are not allowed in {{ ... }}.   Hope this is acceptable...